subsys/net/ip/utils.c

```diff
@@ -496,3 +503,13 @@ u16_t net_calc_chksum_ipv4(struct net_pkt *pkt)
 	return sum;
 }
 #endif /* CONFIG_NET_IPV4 */
+
+bool net_check_header(struct net_pkt *pkt, u8_t *hdr, size_t hdr_size)
```

Comment:
Please suggest better name, I ran out of ideas for a better name when writing the code :)